### PR TITLE
⚡ Refactor build chain and fix imports.

### LINKED ISSUE #30: Error: Must use import to load ES Module
Hi! 
Sorry if I made a stupid mistake. My knowledge in node.js is weak :( But I don’t understand why  doesn’t see module build in your package. 
I am getting an error in the console:
image

Node v16.5.0
bitecs v0.3.15-1

package.json:
image

tsconfig:
image

### PR DESCRIPTION
**Pull request by SupremeTechnopriest**
- Replaces rollup with esbuild
- Adds a build summary to the end of build chain
- Adds `npm run watch` to watch build javascript
- Adds `npm run twatch` to watch build javascript and typedefs
- Fixes #30

### CODE REVIEW

**Comment by SupremeTechnopriest** [comment]
@NateTheGreatt This is the first step to the typescript port and also fixes the import bug.  Josh from XREngine needs this ASAP so if you can review, merge and publish when you have a chance I'm sure he would appreciate it :)  

I also added some watch commands `npm run watch` for js only and `npm run twatch` for js and types.  We wont really need to use these until we get everything ported to typescript, but the build chain is fully ready for fast typescript development. I'm going to create a new branch tomorrow and start the typescript port with this branch as the base.

**Comment by SupremeTechnopriest** [comment]
I noticed that the tests were failing because they didn't have the `.mjs` extension.  I added made the default module type `module` in `package.json` so we don't have to change the extensions.  This allowed me to revert the scripts back to the `.js` extension.  

I updated the npm commands and did some esm import and cjs import tests and everything is all good.  I also did some typescript import tests with all combinations of `ES5` and `ESNext` targets and `CommonJS` and `ESNext` modules.  Also all good.  Automated github actions are passing now. 

This is good to merge.

**Comment by NateTheGreatt** [comment]
many thanks techno 🙏

File: scripts/logLogo.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  [o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o],
+  [o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o],
+  [o, b, o, o, o, b, o, o, o, o, o, b, b, b, o, o, b, b, o, o, o, b, b, b, o],
+  [o, b, o, o, o, o, o, o, b, o, o, b, o, o, o, b, o, o, b, o, b, o, o, o, o],
+  [o, b, b, b, o, b, o, b, b, b, o, b, b, o, o, b, o, o, o, o, o, b, b, o, o],
+  [o, b, o, b, o, b, o, o, b, o, o, b, o, o, o, b, o, o, b, o, o, o, o, b, o],
+  [o, b, b, b, o, b, o, o, b, o, o, b, b, b, o, o, b, b, o, o, b, b, b, o, o],
+  [o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o],
+  [o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o, o]
+]
```
Comment (NateTheGreatt) [inline scripts/logLogo.js:22]: ❤️

**Review by NateTheGreatt — APPROVED** [review]

**Pull request merged**